swarm/src/lib.rs

```diff
@@ -611,9 +613,16 @@ where TBehaviour: NetworkBehaviour,
       TConnInfo: ConnectionInfo<PeerId = PeerId> + fmt::Debug + Clone + Send + 'static,
 {
     pub fn new(transport: TTransport, behaviour: TBehaviour, local_peer_id: PeerId) -> Self {
+        let executor = ThreadPoolBuilder::new()
```

Comment:
This could be moved to the `build` method, then you wouldn't have to pay the cost of constructing a `ThreadPoolBuilder` if the user overwrites it with `executor` or `executor_fn` anyway.